Lib/test/test_tools/test_makefile.py

```diff
@@ -43,7 +43,11 @@ def test_makefile_test_folders(self):
         used = [idle_test]
         for dirpath, _, _ in os.walk(support.TEST_HOME_DIR):
             dirname = os.path.basename(dirpath)
-            if dirname == '__pycache__':
+            if (
+                dirname == '__pycache__'
+                or dirname.startswith('.')
+                or not os.listdir(dirpath)
```

Comment:
Maybe skip not only empty directories, but directories without files (with only subdirectories)? Or even directories without non-hidden files?

Reply:
Also, since we want to skip not only hidden or `__pycache__` directories, we should clear the list of subdirectories to prevent processing them.

```py
if dirname == '__pycache__' or dirname[:1] == '.':
    dirs.clear()
    continue
if all(name[:1] == '.' for name in files):
    continue
```

Reply:
Done! One extra test case:

```bash
mkdir Lib/test/test_hidden                                                                       
touch Lib/test/test_hidden/.hidden 
```